File: back/swift/app/core/service/gameLogic/GameStateChecker.swift
import Foundation

protocol GameStateChecker {
    func checkForVictoryOrDrawA(cells: [[CellType?]]) -> Result<GameResult>;
    func checkForVictoryOrDrawB(cells: [CellPosition: CellType?]) -> Result<GameResult>;
    func checkForVictoryOrDrawC(playersHands: [CellType: [CellPosition]]) -> Result<GameResult>;
}

File: back/kotlin/app/core/service/gameLogic/GameStateChecker.kt
interface GameStateChecker  {
    fun checkForVictoryOrDrawA(cells: Array<Array<CellType?>>): Result<GameResult>;
    fun checkForVictoryOrDrawB(cells: Map<CellPosition, CellType?>): Result<GameResult>;
    fun checkForVictoryOrDrawC(playersHands: Map<CellType, List<CellPosition>>): Result<GameResult>; 
}